### ISSUE TITLE
cluster_pixels function fails to delete pixel_mat_data folder

### ISSUE THREAD

**Issue by HPiyadasa** [angelolab/ark-analysis#936]
**Describe the bug**
Cluster_pixels function works as intended on a test set of 28 images. But when using a larger set of data (604 FOVs), the following error is generated after all 604 FOVs have been processed (attempted this 2 times to confirm) 

" [Errno 2] No such file or directory: '/Volumes/Elite Pro/DCIS/Cohorts/20230209_DCIS/pixie/Round_1_attemp_2_pixel_output_dir/pixel_mat_data/._TA583_R8C5.feather'"

<img width="1568" alt="Screenshot 2023-02-22 at 10 21 31 AM" src="https://user-images.githubusercontent.com/62519486/220690779-34d2ac2d-acc2-467c-bdf6-cd5e2897e06c.png">

Upon inspecting the data directory -  the pixel_mat_data and pixel_mat_data_temp are both still present. However, in pixel_mat_data folder there are only 603 FOVs (TA583_R8C5.feather file indicated in the error above has been deleted) while in pixel_mat_data_temp folder there are the 604 FOVs. 

<img width="743" alt="Screenshot 2023-02-22 at 10 26 31 AM" src="https://user-images.githubusercontent.com/62519486/220690896-ee6e3e02-cf3a-48bc-9118-9b97b53f5346.png">

<img width="733" alt="Screenshot 2023-02-22 at 10 27 17 AM" src="https://user-images.githubusercontent.com/62519486/220692124-5d8ddcbc-cd45-4d88-a620-ae2def31ebbf.png">

<img width="735" alt="Screenshot 2023-02-22 at 10 27 48 AM" src="https://user-images.githubusercontent.com/62519486/220692177-efee7835-3dec-4c91-a707-341340e4a837.png">

**Comment by HPiyadasa** [comment]
@cliu72 helped me figure out the issue. The issue and the solution provided here works 

https://stackoverflow.com/questions/64687047/shutil-rmtree-filenotfounderror-errno-2-no-such-file-or-directory-xxx

TLDR - 
" Mac filesystem drivers automatically create binary companion files prefixed with ._ to record so-called extended attributes on non HFS+ file systems (external HDD). rmtree on systems which do not support file descriptors in os.scandir (like Mac OSX) now unsafely creates a list of entries and then unlinks them one by one"

Issue - 

1. the original file is always listed before the extended attributes one, and
2. when the original file is unlinked (test.txt) the meta file (._test.txt) is removed simultaneously.

Thus, the extended attribute file will be missing when it is its turn and throw the FileNotFoundError"

**Mitigation**

def ignore_extended_attributes(func, filename, exc_info):
    is_meta_file = os.path.basename(filename).startswith("._")
    if not (func is os.unlink and is_meta_file):
        raise

shutil.rmtree(path_dir, onerror=ignore_extended_attributes)

**This fixed the issue**
